src/app/components/did/manage-dids/manage-dids.component.ts

```diff
@@ -92,8 +97,22 @@ export class ManageDidsComponent implements OnInit {
 
   closeFormScreen() {
     this.formScreenOpen = false;
-    this.didDocuments = this.vaultService.getAllDIDDocuments();
-    this.didIds = Object.keys(this.didDocuments);
+    this.getDIDsInfo();
+  }
+
+  search(searchTerm: string) {
+    this.didIds = [];
+    for (const didId in this.allDIDsPublicInfo) {
+      if (didId.includes(searchTerm)) {
+        this.didIds.push(didId);
+      } else if (this.allDIDsPublicInfo[didId].nickname.includes(searchTerm)) {
+        this.didIds.push(didId);
+      }
+    }
```

Comment:
What exactly is inside `didId`: is it the "did:factom:[32 bytes]"? If yes, then I don't think it makes sense to allow searching over it. We should instead only support the search over the human-readable nickname.